### ISSUE TITLE
e2e tests logs are missing the entire controller manager logs 

### ISSUE THREAD

**Issue by jsturtevant** [kubernetes-sigs/cluster-api-provider-azure#1876]
/kind bug

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
The job https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-periodic-e2e-full-main is missing the full set of controller manager logs. If you look for vmss in 

- https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1461402439272894464/artifacts/clusters/bootstrap/controllers/capi-controller-manager/capi-controller-manager-6bdc78c4d4-nnxc7/manager.log 
- https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1461402439272894464/artifacts/clusters/bootstrap/controllers/capz-controller-manager/capz-controller-manager-5f7b59585c-m9lxd/manager.log 
 
you will not find relevant logs for that cluster even though the cluster was created: https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1461402439272894464/artifacts/clusters/capz-e2e-4e4w9c-win-vmss/machines/

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

My guess is the logs are rolling and we are missing the first part of the logs

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by jsturtevant** [comment]
I guess this currently is not in capz specifically. It looks like it is provided by the capi framework: https://github.com/kubernetes-sigs/cluster-api/blob/aafa086da595450ff1f75d4a29b213c873f15e4d/test/framework/clusterctl/clusterctl_helpers.go#L103-L109

**Comment by jsturtevant** [comment]
So looking closer It seems like there maybe the stream is dropped at some point or restart of the agent logs and the stream doesn't re-establish itself.  Example:

Last time logged in [capz log](https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1461402439272894464/artifacts/clusters/bootstrap/controllers/capz-controller-manager/capz-controller-manager-5f7b59585c-m9lxd/manager.log): `I1118 18:58:03.945938  1 composite_logger.go:44] `
time in [build file](https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1461402439272894464/build-log.txt): `Nov 18 19:33:27.844: INFO: Creating cluster identity secret`

**Comment by jsturtevant** [comment]
not exactly the same but does call this out: https://github.com/kubernetes-sigs/cluster-api/issues/4684

**Comment by jsturtevant** [comment]
The [capi controller ](https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-full-main/1461402439272894464/artifacts/clusters/bootstrap/controllers/capi-controller-manager/capi-controller-manager-6bdc78c4d4-nnxc7/manager.log)also stops at same time: `E1118 18:58:03.922445       1 controller.go:317] controller/cluster`

**Comment by jsturtevant** [comment]
this still occurs in our e2e tests that create multiple clusters 
/remove-lifecycle stale

**Comment by CecileRobertMichon** [comment]
@jsturtevant I think I tracked this down

**Comment by jackfrancis** [comment]
@CecileRobertMichon @willie-yao is this still an issue?

**Comment by CecileRobertMichon** [comment]
From what I can remember, it's not completely fixed but it's improved. All tests run in parallel now so we at least get some logs for all clusters but we still have the issue where the first test that returns interrupts log collection and we loose logs for the ones still running.

**Comment by willie-yao** [comment]
@jackfrancis As a side note for developers looking to run the E2E tests locally but don't want to wait for all the logs to timeout, I added a `skipLogCollection` flag to the E2E test inputs in #2687

**Comment by willie-yao** [comment]
Whoops I thought I was on a separate issue, disregard my last comment. I didn't work on this directly and thought we were referencing the issue with log collection timing out.

**Comment by CecileRobertMichon** [comment]
/remove-lifecycle stale
/priority-important-longterm
/unassign

I'm not actively working on this right now but we should really figure this out at some point

**Comment by bhanuprakashgrb** [comment]
Hi maintainers 👋

I’m interested in helping with this issue.

From the discussion, it looks like the missing logs may be related to log streaming or rotation in the Cluster API test framework rather than CAPZ-specific logic.

Before starting, I wanted to confirm: would it be useful to begin by adding additional diagnostics or safeguards around controller-manager log collection in the e2e framework (for example detecting stream restarts or capturing earlier logs)?

Happy to start small and iterate based on feedback

**Comment by nojnhuh** [comment]
I forgot we had an issue open for this, but I recently fixed this in #6073.

/close
